### PR TITLE
Frontend Prep and Updates

### PR DESCRIPTION
**Pull request by davidshq**
# Upgrade Dependencies
- Vue is upgraded from 2.6.14 to 2.7.14 which integrates the Vue Composition API directly into Vue.
- Vue Router, Vue Template Compiler, Vuetify Loader also see updates.
- Vue Composition API as a standalone dependency is removed.
- An overrides setting is added to the `package.json` so that using TypeScript 5.x doesn't cost `npm i` to fail.
  - There is a known bug in npm which prevents overrides from working correctly with existing `node_modules` and `package-lock.json` files, if running into issues remove these and retry.
- Update code in existing composition components to work with 2.7.x.
- Update tsconfig.json so latest Volar tools work optimally with Vue code.

# ControlPanel.vue
- Refactored to utilize composition instead of options API.

# TypeScript
- A number of smaller changes made throughout that fix issues caught by upgraded TS, ESLint, etc.

# Vuex
- Remove remaining dependencies on `direct-vuex`.

# Other
- Fixes to Vue shim
- Reorder types, add additional

### CODE REVIEW

File: web_client/.eslintrc.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ module.exports = {
   },
   extends: [
     'plugin:vue/recommended',
-    'plugin:vuetify/recommended',
+    // 'plugin:vuetify/recommended',
```
Comment (annehaley) [inline web_client/.eslintrc.js:8]: Can you explain this change?

File: web_client/.github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Playwright Tests
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
```
Comment (annehaley) [inline web_client/.github/workflows/playwright.yml]: I've found that specifying both `push` and `pull_request` here will run the tests twice if you push to a PR. Just `push` will suffice.

File: web_client/src/components/CustomRangeSlider.vue [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default VRangeSlider.extend({
       } else {
         this._middleDragStart = null;
       }
-      VRangeSlider.options.methods.onSliderMouseDown.call(this, e);
+      (VRangeSlider as any).options.methods.onSliderMouseDown.call(this, e);
```
Comment (annehaley) [inline web_client/src/components/CustomRangeSlider.vue]: I see an error saying `unexpected any`. Can this be cast to an html element or something similar? There are multiple warnings about these raised by GH actions, you can see them in the Files Changed tab.

File: web_client/tests/basic-test.spec.ts-snapshots/test-1-chromium-linux.png
Comment (annehaley) [inline web_client/tests/basic-test.spec.ts-snapshots/test-1-chromium-linux.png:1]: Was this file intended to be added?

File: web_client/tests/basic-test.spec.ts
Comment (annehaley) [inline web_client/tests/basic-test.spec.ts:1]: This looks like an awesome test. This is what we were trying to accomplish with the pyppeteer testing, but you may have noticed that the pyppeteer tests are not deterministic nor reliable. If these tests prove to be better, we should consider completely replacing the pyppeteer ones.

**Review by annehaley — CHANGES_REQUESTED** [review]
I'm loving the new composition api look for this code. Thank you for putting in that work; it's something we avoided for too long.

The playwright tests addition seems like a good idea. I see what looks like browser compatibility testing. Is there a reason we only have the chromium one uncommented? 

Only saw a few minor things to address.

File: web_client/.eslintrc.js (hunk 1 above)
Comment (davidshq) [reply web_client/.eslintrc.js:8]: This is temporary. Vuetify was causing the linting to break b/c it was throwing errors and modifying the code to comply with Vuetify's recommendations would cause significant layout shift. I've made the changes on a branch I've got but I think we should usher the changes in more slowly (and with some more tests) to ensure we understand the impact.

File: web_client/src/components/CustomRangeSlider.vue (hunk 3 above)
Comment (davidshq) [reply web_client/src/components/CustomRangeSlider.vue]: It wouldn't take as an `HTMLElement` but it appears if I import `VRangeSlider` with an explicit path TS is able to grab the typing.

File: web_client/tests/basic-test.spec.ts-snapshots/test-1-chromium-linux.png
Comment (davidshq) [reply web_client/tests/basic-test.spec.ts-snapshots/test-1-chromium-linux.png:1]: Yes, actually! This is the "snapshot of truth" when it comes to the basic-test. It is the standard against which snapshots we might make performing local runs of the test will be compared against.

If we make a change and we want the UI to be updated then we can run `--update-snapshots` to replace this image with a new one that becomes the base line.

I have a few notes on Playwright here: https://github.com/davidshq/miqa-notes/blob/main/annotations-project/how-to/vue-testing/playwright-testing.md

File: web_client/tests/basic-test.spec.ts
Comment (davidshq) [reply web_client/tests/basic-test.spec.ts:1]: The author of pyppeteer actually recommends folks move to Playwright and has stopped updating pyppeteer. It also looks like we can probably port a lot of the test code over pretty seamlessly: https://playwright.dev/docs/puppeteer

Do you have a way to get the pyppeteer tests up and running in this repo? They seem to be failing every time and I don't think it is due to code changes I've made. If not, maybe we should disable short-term and plan to migrate to playwright down the road?

File: web_client/.github/workflows/playwright.yml (hunk 2 above)
Comment (davidshq) [reply web_client/.github/workflows/playwright.yml]: Will do!

**Comment by davidshq** [comment]
Re: cross-browser testing

I agree it would be better if we ran the tests cross-browser - but it does take around 3x as long as running through a single test. This isn't a big deal in the CI but it starts to get onerous when doing local dev. That said, there may be a way to configure it so that it runs single browser on local and cross-browser for CI.

This is the first project I'm using Playwright on, so it's likely I don't know all the ins-and-outs and this situation seems like a common scenario.

**Comment by annehaley** [comment]
Disabling the pyppeteer tests and running a playwright test for one browser is acceptable for now. Could you make this change and make an issue to address the browser testing migration in the future?

**Review by annehaley — APPROVED** [review]

**Comment by annehaley** [comment]
@davidshq Do you mind if I merge this? I'd like to propagate it to the main repo.

**Comment by davidshq** [comment]
Go for it!
